runtime/vm/ContinuationHelpers.cpp

```diff
@@ -1227,8 +1227,14 @@ takeVirtualThreadListToUnblock(J9VMThread *currentThread)
 					break;
 				}
 				default:
-					/* Thread must be unblocked by AfterYield(), hence it can only be UNBLOCKED or RUNNING. */
-					Assert_VM_true((JAVA_LANG_VIRTUALTHREAD_RUNNING == state) || (JAVA_LANG_VIRTUALTHREAD_UNBLOCKED == state));
+				{
+					if (J9_ARE_ANY_BITS_SET(state, JAVA_LANG_VIRTUALTHREAD_SUSPENDED)) {
+						J9VMJAVALANGVIRTUALTHREAD_SET_BLOCKPERMIT(currentThread, current->vthread, JNI_TRUE);
+					} else {
+						/* Thread must be unblocked by AfterYield(), hence it can only be UNBLOCKED or RUNNING. */
+						Assert_VM_true((JAVA_LANG_VIRTUALTHREAD_RUNNING == state) || (JAVA_LANG_VIRTUALTHREAD_UNBLOCKED == state));
+					}
+				}
```

Comment:
`break` is needed for the `default` case if the assertion is not triggered. to differentiate between the default and switch case brackets since they are at the same level ... i would add a comment
```suggestion
						Assert_VM_true((JAVA_LANG_VIRTUALTHREAD_RUNNING == state) || (JAVA_LANG_VIRTUALTHREAD_UNBLOCKED == state));
					}
					break;
				}
				} /* switch (state) */
```

Reply:
updated